python/pyspark/cloudpickle.py

```diff
@@ -40,164 +40,126 @@
 NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 """
-
+from __future__ import print_function
 
 import operator
 import os
+import io
 import pickle
 import struct
 import sys
 import types
 from functools import partial
 import itertools
-from copy_reg import _extension_registry, _inverted_registry, _extension_cache
-import new
 import dis
 import traceback
-import platform
-
-PyImp = platform.python_implementation()
-
 
-import logging
-cloudLog = logging.getLogger("Cloud.Transport")
```

Comment:
It's no longer vendoring when changes are happening in your own code base of cloudpickle. This ends up being even worse for projects hoping to use it when pyspark itself isn't pip installable either.

What's the best path forward for us to help maintain cloudpickle in a way that is friendly to you vendoring it?